changelog.d/8345.feature

```diff
@@ -0,0 +1 @@
+Add a configuration option that allows existing users to log in with OpenID Connect, and fix a type hints error. Contributed by @BBBSnowball and @OmmyZhang.
```

Comment:
I think we can leave the type hints fix out -- people running Synapse are likely less interested in that and those are the sort of bugs we fix as we find them doing other work!

```suggestion
Add a configuration option that allows existing users to log in with OpenID Connect. Contributed by @BBBSnowball and @OmmyZhang.
```